File: solutions/go/rna-transcription/2/rna_transcription.go
package strand

import "strings"

var dnaToRnaMap = map[rune]rune{
	'A': 'U',
	'C': 'G',
	'G': 'C',
	'T': 'A',
}

func ToRNA(dna string) (rna string) {
	return strings.Map(func(n rune) rune { return dnaToRnaMap[n] }, dna)
}

File: solutions/ocaml/rna-transcription/2/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let transcribe_nucleotide = function
  | `A -> `U
  | `C -> `G
  | `G -> `C
  | `T -> `A

let to_rna = List.map transcribe_nucleotide
